src/OSPSuite.Infrastructure.Import/Core/DataFormat/DataFormatHeadersWithUnits.cs

```diff
@@ -33,6 +34,7 @@ protected override UnitDescription ExtractUnits(string description, IUnformatted
             .Substring(1, units.Length - 2) //remove the brackets
             .Trim() //remove whitespace
             .Split(',') //split comma separated list
+            .Where(unitName => supportedDimensions.Any(x => x.HasUnit(unitName))) //only accepts valid and supported units
```

Comment:
I don't understand this part and it seems to me it will not work if the caption has more then one [] 
E.g. what would the unit be for `"[min] residence time [h]"`? Should be `h`

Reply:
Actually for this case, what will be recognized as a unit will be everything that is between the first [ and the last ] ( so in this example "min] residence time [h" ), and thus the unit will not be automatically detected. This means that if somebody tries to import a file with a header "Time [min] residence time [h]", the initial mapping for time will be invalid unit, and the user should fix it manually. Personally I do not see this as that big a problem.  But if you believe it should be adjusted I can do a new issue for this.

In any case, this was like that also before the fix. All thsi added .where() clause does, is making suire that the initial mapping does not assign a unit to a mapping that does not belong to one of its supported mappings: a header with "Concentration[s]" will result in the initial unit for concentration being set to invalid unit. Before this change, concentration would be assigned initially assigned unit "s" and dimension "time" - something that should not be possible.

Reply:
how about this? Just checks if the (trimmed) caption ends with `[something]` and takes `something `as unit (if it's supported): 
```
if (!name.Trim().EndsWith("]") || name.LastIndexOf("[")<0)
    return new UnitDescription();
var unit = name.Substring(name.LastIndexOf("[") + 1, name.LastIndexOf("]") - name.LastIndexOf("[") - 1);

if (!supportedDimensions.Any(x => x.HasUnit(unit)))
  unit = UnitDescription.InvalidUnit;
         ...
```
Results in:
- "Time [h]" => h
- "[min] Time [h]" =>h
- "Time [blabla] [h]"=> h
- "[Time after dose] [h]" => h
- "[min] Time"=> InvalidUnit
- "Time [semester]" => InvalidUnit
etc.

Reply:
I mean.....we could of course implement it like that. But do you really think this is a good timing to change something like this? Is it really a plausible scenario to have more than one pair of brackets in the header? If we need this, let's create a new issue for this and I will implement it.

Reply:
If it's sufficiently covered by tests - why not?
Also as you already said it's just the **initial** mapping - so I don't see any dangerous side effects.
And having the ß-test phase in front - I would risk the change :)

Reply:
@georgeDaskalakis I guess you are still working on this?

Reply:
yes, but in this new issue: https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/1217